src/main/java/core/dao/DataToStringImpl.java

```diff
@@ -0,0 +1,13 @@
+package core.dao;
+
+import core.db.Storage;
+import java.util.stream.Collectors;
+
+public class DataToStringImpl implements DataToString {
+    @Override
+    public String generateString() {
+        return Storage.fruits.entrySet().stream()
+                .map(s -> s.getKey() + "," + s.getValue())
+                .collect(Collectors.joining(System.lineSeparator()));
```

Comment:
this method should also be in the FruitService, so you may want use the dao layer here as well instead of directly addressing the storage